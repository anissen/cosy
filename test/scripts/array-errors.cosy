{
    let arr = ['5','6','7']
    arr.push('8') // fails because arr is not mut

    fn array_push(a Array Str) {
        a.push('9') // fails because a is not mut
        print a
    }
    array_push(arr)
    print arr
}


{
    mut arr = ['2','3','4']
    arr[1] += 3
    print arr
}

{
    mut arr = [2, 3, 4]
    arr[1] += '3'
    print arr
}