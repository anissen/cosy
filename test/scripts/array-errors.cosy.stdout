■ test/scripts/array-errors.cosy, line 3:
1 | {
2 |     let arr = ['5','6','7']
3 |     arr.push('8') // fails because arr is not mut
            ^^^^ Cannot call mutating method on immutable array.
4 | 
5 |     fn array_push(a Array Str) {

■ test/scripts/array-errors.cosy, line 6:
4 | 
5 |     fn array_push(a Array Str) {
6 |         a.push('9') // fails because a is not mut
              ^^^^ Cannot call mutating method on immutable array.
7 |         print a
8 |     }

■ test/scripts/array-errors.cosy, line 16:
14 | {
15 |     mut arr = ['2','3','4']
16 |     arr[1] += 3
                ^^ Expected value of type Str but got Num
17 |     print arr
18 | }

■ test/scripts/array-errors.cosy, line 22:
20 | {
21 |     mut arr = [2, 3, 4]
22 |     arr[1] += '3'
                ^^ Expected value of type Num but got Str
23 |     print arr
24 | }

